amundsen_application/static/js/ducks/bookmark/reducer.ts

```diff
@@ -135,15 +143,21 @@ export default function reducer(
         },
       };
     case GetBookmarksForUser.SUCCESS:
+      const bookmarksForUserResponse = <GetBookmarksForUserResponse>action;
```

Comment:
@Golodhros this PR pokes the beehive by revealing some latent non-null checks, and I'm trying to fix here.

I'm pretty uncertain of how clean this approach. I think it would be ideal to determine success higher up (perhaps using status codes), then you either have a fully-complete response with payload, or you have a failure. Having the presence of certain keys be dependent on other keys seems like a parsing mess, and while this certainly pleases the type-checker, it feels pretty fraught with errors. Definitely too big to fix in this PR, but is this the right temporary fix?

Reply:
I think it looks good for now. 
We have plenty of these being tracked in our betterer, hopefully we'll get to them soon!

Reply:
great!